core/src/main/java/com/sopt/core/designsystem/component/dialog/NoostakDialog.kt

```diff
@@ -0,0 +1,154 @@
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.Card
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.RectangleShape
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.window.Dialog
+import androidx.compose.ui.window.DialogProperties
+import com.sopt.core.R
+import com.sopt.core.designsystem.theme.NoostakAndroidTheme
+import com.sopt.core.designsystem.theme.NoostakTheme
+import com.sopt.core.type.DialogType
+
+@Composable
+fun NoostakDialog(
+    dialogType: DialogType,
+    onClick: () -> Unit = {},
+    onDismissRequest: () -> Unit = {}
+) {
+    Dialog(
+        onDismissRequest = { onDismissRequest() },
+        properties = DialogProperties(
+            dismissOnBackPress = true,
+            dismissOnClickOutside = true
+        )
+    ) {
+        Card(
+            shape = RoundedCornerShape(dimensionResource(id = R.dimen.dialog_radius))
+        ) {
+            Column(
+                modifier = Modifier
+                    .width(274.dp)
+                    .wrapContentHeight()
+                    .background(
+                        color = NoostakTheme.colors.white
+                    ),
+                horizontalAlignment = Alignment.CenterHorizontally
+            ) {
+                Spacer(
+                    modifier = Modifier.height(
+                        when (dialogType) {
+                            DialogType.GROUP -> 24.dp
+                            DialogType.LOGOUT -> 41.dp
+                            DialogType.WITHDRAWAL -> 30.dp
+                        }
+                    )
+                )
+                Text(
+                    text = stringResource(dialogType.content),
+                    textAlign = TextAlign.Center,
+                    style = if (dialogType == DialogType.GROUP) NoostakTheme.typography.c3Regular else NoostakTheme.typography.b4Regular
+                )
+                Spacer(
+                    modifier = Modifier.height(
+                        when (dialogType) {
+                            DialogType.GROUP -> 20.dp
+                            DialogType.LOGOUT -> 36.dp
+                            DialogType.WITHDRAWAL -> 26.dp
+                        }
+                    )
+                )
+                Box(
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .height(1.dp)
+                        .background(color = NoostakTheme.colors.gray200)
+                )
```

Comment:
P1: 선은 박스 대신에 divider 사용합시당
```suggestion
                HorizontalDivider(
                    modifier = Modifier.fillMaxWidth(),
                    color = NoostakTheme.colors.gray200,
                    thickness = 1.dp
                )
```

Reply:
하하 긁어온거 들켰네요